Include/internal/pycore_stackref.h

```diff
@@ -167,7 +167,7 @@ PyStackRef_FromPyObjectNew(PyObject *obj)
 // Converts a PyStackRef back to a PyObject *, converting deferred references
 // to new references.
 static inline PyObject *
-PyStackRef_AsPyObjectNew(_PyStackRef tagged)
+PyStackRef_AsPyObjectDeferredToNew(_PyStackRef tagged)
```

Comment:
What's the semantics of this?
Does the stack ref remain valid after the call?

#### Yes
If it does and since the caller doesn't, in general, know whether `tagged` is deferred or not, this may or may not create a new reference.
In other words it appears that this is equivalent to:
```C
    obj = PyStackRef_AsPyObjectBorrow(ref);
    if (random) Py_INCREF(obj);
```
#### No
Then how does this differ from `PyStackRef_AsPyObjectSteal`?

Reply:
The equivalent is the following:
```
    obj = PyStackRef_AsPyObjectSteal(ref);
    if (is_deferred_object) Py_INCREF(obj);
```
This is required because objects in the ZCT (or stack in this matter), has a refcount of 0. If we pass the object to something that doesn't understand that, it will segfault. So we need to incref it before passing it.

Reply:
But how does the caller know if an object is deferred?
`is_deferred_object` is indisinguishable from `random.choice((False, True))`

Reply:
Sam sent a PR that marks an object is deferred in the GC bits. A future version of this PR will tag those objects. I didn't turn that on for now because it requires stack scanning.

Reply:
`is_deferred_object` = `_PyObject_HasDeferredRefcount` (if we defer everything like in your scheme) OR
`is_deferred_object` = `tag & 0b1` (if we defer just a few things like in PEP 703)

Reply:
If I call `PyStackRef_AsPyObjectDeferredToNew` do I own the return reference or not?

This should have nothing to do with tag bits, ZCT or any other implementation details.

Reply:
Ohh I get the question now. Sorry you were talking about the semantics. Yes I own the reference now after calling it.

Reply:
So it is the same as `PyStackRef_AsPyObjectNew` then?

Reply:
This thread suggests I should rename it to `PyStackRef_AsPyObjectDeferredToSteal`

Reply:
I think the "Deferred" part should be dropped. It isn't part of the API, it is an implementation detail.
The only API we should need is `Borrow` and `Steal` and, if it keeps the code cleaner, a `New` variant as well.
i.e.
```C
PyStackRef_AsPyObjectBorrow() /* Try to avoid this, as it makes tagging harder */
PyStackRef_AsPyObjectSteal()
PyObject_AsPyStackRefBorrow() /* Try to avoid this, as it makes tagging harder */
PyObject_AsPyStackRefSteal()
```
and maybe
```C
PyStackRef_AsPyObjectNew()
PyObject_AsPyStackRefNew()
```